### ISSUE TITLE
maths manipulations use new_data_container_handle

### ISSUE THREAD

**Issue by rijobro** [CCPPETMR/SIRF#394]
The current way for doing maths in python and matlab is along the lines of:
```
void* h = x.new_data_container_handle();
DataContainer& z = objectFromHandle<DataContainer>(h);
z.axpby(ptr_a, x, ptr_b, y);
return h;
```
This works with STIR, for example, because `new_data_container_handle` actually returns a clone. I don't think `new_data_container_handle` should return a clone, but should instead return a blank copy of itself. My NIfTI stuff returns a blank copy, which then causes the third line to crash (because none of the data has actually been allocated).

Instead, I think the above code would be more correct as:
```
std::shared_ptr<DataContainer > z_sptr = x.clone();
z_sptr->axpby(ptr_a, x, ptr_b, y);
return newObjectHandle<DataContainer >(z_sptr);
```

@evgueni-ovtchinnikov , @KrisThielemans - thoughts?
If you agree with my line of logic, then we should also change the implementation of `new_data_container_handle` to return a blank copy instead of a clone.

**Comment by evgueni-ovtchinnikov** [comment]
`new_data_contaner_handle` returns a handle to an empty container, not to a clone

in your example this new empty container (`z`) is filled by its method `axpby`

**Comment by rijobro** [comment]
> new_data_contaner_handle returns a handle to an empty container, not to a clone

Apologies, it's not a clone, but it's an empty image with the same number of voxels, which have already been allocated. Here's the STIR implementation:
```
virtual ObjectHandle<DataContainer>* new_data_container_handle() const
{
	return new ObjectHandle<DataContainer>
		(stir::shared_ptr<DataContainer>(same_image_data()));
}
```
And `same_image_data` looks like this:
```
STIRImageData* same_image_data() const
{
	STIRImageData* ptr_image = new STIRImageData;
	ptr_image->_data.reset(_data->get_empty_copy());
	ptr_image->set_up_geom_info();
	return ptr_image;
}
```
Is this really the functionality that we want? From the name alone, it doesn't seem intuitive to me that `new_data_container_handle` should actually return a blank copy of the original image with the same number of voxels (that have already been allocated).

Maybe we could just change names? i.e., 
    - `same_image_data` -> `get_empty_copy` (as per STIR)
    - `new_data_container_handle` -> `get_empty_copy_container_handle`

**Comment by evgueni-ovtchinnikov** [comment]
sorry, what i wrote was about acquisition data

indeed, for STIR images it is a copy, but not for Gadgetron's

**Comment by rijobro** [comment]
Although I still find the names unobvious, I solved my particular problem with #396.
